From c1cca58f36547957461892cea88e6f22297560f3 Mon Sep 17 00:00:00 2001
From: Richard Fitzgerald <rf@opensource.wolfsonmicro.com>
Date: Mon, 1 Sep 2014 15:52:04 +0100
Subject: [PATCH 312/351] mfd: arizona: Remove bogus warning in
 arizona_dvfs_down

The flags are simple booleans per block, not reference counts.
Each block that has a DVFS dependency is not itself reference
counted, it's a simple either/or case which these flags match.
Because the requestors don't reference count themselves we
allow them to notify the same state that they are already in.

Change-Id: I3ea6dbf3f9e265cfed0d3ca4b27ea4f757c90d6a
Signed-off-by: Richard Fitzgerald <rf@opensource.wolfsonmicro.com>
---
 drivers/mfd/arizona-core.c |    3 ---
 1 file changed, 3 deletions(-)

diff --git a/drivers/mfd/arizona-core.c b/drivers/mfd/arizona-core.c
index b9699b8..37b85b8 100644
--- a/drivers/mfd/arizona-core.c
+++ b/drivers/mfd/arizona-core.c
@@ -149,9 +149,6 @@ int arizona_dvfs_down(struct arizona *arizona, unsigned int flags)
 
 	mutex_lock(&arizona->subsys_max_lock);
 
-	if ((arizona->subsys_max_rq & flags) != flags)
-		dev_warn(arizona->dev, "Unbalanced DVFS down: %x\n", flags);
-
 	arizona->subsys_max_rq &= ~flags;
 
 	if (arizona->subsys_max_rq == 0) {
-- 
1.7.9.5

